src/cli/telemetry/helpers.ts

```diff
@@ -0,0 +1,25 @@
+import { TERMINAL_INFO } from '@ionic/utils-terminal';
+
+export const tryFn = async <T extends (...args: any[]) => Promise<R>, R>(
+	fn: T,
+	...args: any[]
+): Promise<R | null> => {
+	try {
+		return await fn(...args);
+	} catch {
+		// ignore
+	}
+
+	return null;
+};
+
+export const isInteractive = (): boolean => TERMINAL_INFO.tty && !TERMINAL_INFO.ci;
+
+export function uuidv4(): string {
+	return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
```

Comment:
Is there a strong need to roll our own UUID v4 generation? I suppose Stencil prefers a "zero dependency" approach, which is a good reason to avoid pulling in something like [UUID](https://www.npmjs.com/package/uuid). Do we have any documentation we can attach to this function that explains how we decided on this implementation?

Reply:
I copied this from Ionic-CLI and their approach to the UUID generation, in order to keep that synced up without adding dependencies.

Reply:
Added!

Reply:
FWIW, Node 14 added [crypto.randomUUID](https://nodejs.org/dist/latest-v14.x/docs/api/all.html#crypto_crypto_randomuuid_options).

Reply:
I take this back, we support Node 12 ATM